src/category_theory/limits/kan_extension.lean

```diff
@@ -0,0 +1,370 @@
+    intros x y ff,
+    simp only [functor.comp_map, nat_trans.naturality_assoc, obj_aux_map, category.assoc],
+    congr' 1,
+    erw [limit.pre_π, limit.w (diagram ι F _) (index.mk_hom (𝟙 _) ff)],
+    congr,
+    tidy,
+  end },
+  inv_fun := λ f,
+  { app := λ x, limit.lift (diagram ι F x) (cone _ f),
+    naturality' := begin
+      intros x y ff,
+      ext j,
+      erw [limit.lift_pre, limit.lift_π, category.assoc, limit.lift_π (cone _ f) j],
+      delta cone index.map,
+      tidy,
+    end },
+  left_inv := begin
+    intro x,
+    ext k j,
+    dsimp only [cone, diagram],
+    rw limit.lift_π,
+    simp only [nat_trans.naturality_assoc, obj_aux_map],
+    congr' 1,
+    erw limit.pre_π,
+    congr,
+    cases j,
+    tidy,
+  end,
+  right_inv := by tidy }
+
+/-- A variant of `Ran.equiv` with `whiskering_left` instead of functor composition. -/
+@[simps]
+def equiv' [∀ x, has_limits_of_shape (index ι x) D] (F : S ⥤ D) (G : L ⥤ D) :
+  (G ⟶ obj_aux ι F) ≃ (((whiskering_left _ _ _).obj ι).obj G ⟶ F) := equiv _ _ _
+
+end Ran
+
+/-- The right Kan extension of a functor. -/
+@[simps]
+def Ran [∀ X, has_limits_of_shape (Ran.index ι X) D] : (S ⥤ D) ⥤ L ⥤ D :=
+adjunction.right_adjoint_of_equiv (λ F G, (Ran.equiv' ι G F).symm) (by tidy)
+
+namespace Ran
+
+variable (D)
+
+/-- The adjunction associated to `Ran`. -/
+def adjunction [∀ X, has_limits_of_shape (Ran.index ι X) D] :
+  (whiskering_left _ _ D).obj ι ⊣ Ran ι :=
+adjunction.adjunction_of_equiv_right _ _
+
+end Ran
+
+namespace Lan
+
+local attribute [simp] comma.fst comma.map_right
+
+/-- The index category of limits used to define `Lan`. -/
+@[simp, derive category, nolint has_inhabited_instance]
+def index (l : L) := comma ι (functor.from_punit l)
+
+variable {ι}
+
+/-- Make a term of type `Lan.index ι x`. -/
+@[simp]
+def index.mk {x : L} {y : S} (f : ι.obj y ⟶ x) : index ι x := ⟨y, ⟨⟩, f⟩
+
+/-- The functor `Lan.index ι x ⥤ Lan.index ι y` associated to a morphism `x ⟶ y`. -/
+@[simp]
+def index.map {x y : L} (f : x ⟶ y) : index ι x ⥤ index ι y :=
+comma.map_right _ ((functor.const _).map f)
+
+/-- Make a morphism in `Lan.index ι x`. -/
+@[simps]
+def index.mk_hom {x : L} {y z : S} (f : ι.obj z ⟶ x) (g : y ⟶ z) :
+  index.mk (ι.map g ≫ f) ⟶ index.mk f :=
+{ left := g,
+  right := 𝟙 _,
+  w' := by simpa }
+
+lemma index.map_mk {x y : L} {z : S} (f : ι.obj z ⟶ x) (g : x ⟶ y) :
+  (index.map g).obj (index.mk f) = index.mk (f ≫ g) := rfl
+
+lemma index.map_id {x : L} {j : index ι x} :
+  (index.map (𝟙 x)).obj j = j := by {cases j, tidy}
+
+lemma index.map_comp {x y z : L} (f : x ⟶ y) (g : y ⟶ z) (j : index ι x) :
+  (index.map (f ≫ g)).obj j = (index.map g).obj ((index.map f).obj j) :=
+by {cases j, tidy}
+variable (ι)
+
+-- TODO: Use this to prove that `Lan.adjunction` is coreflective
+-- when `ι` is fully faithful.
+/-- `index.mk (𝟙 (ι.obj y))` is terminal when `ι` is fully faithful. -/
+def index.mk_id_terminal [full ι] [faithful ι] {y : S} : is_terminal (index.mk (𝟙 (ι.obj y))) :=
```

Comment:
```suggestion
lemma index.mk_id_terminal [full ι] [faithful ι] {y : S} : is_terminal (index.mk (𝟙 (ι.obj y))) :=
```
I have a suspicion that this result might be useful outside of Kan extensions - perhaps it could go in `limits/comma`?

Reply:
Marking this as resolved since it's covered in #6830